src/Chainweb/RestAPI/Backup.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | An endpoint for making database backups. Also used to synchronize 
+-- databases between nodes.
+
+module Chainweb.RestAPI.Backup
+    ( BackupApi
+    , someBackupApi
+    , someBackupServer
+    ) where
+
+import Control.Concurrent.Async
+import Control.Monad.Catch
+import Control.Monad.IO.Class
+import Data.Proxy
+import Data.Text (Text)
+import qualified Data.Text as T
+import Servant
+
+import qualified Chainweb.Backup as Backup
+import Chainweb.Logger
+import Chainweb.Time
+
+import Chainweb.RestAPI.Utils
+
+type BackupApi 
+  =    "make-backup" :> Get '[PlainText] Text
+  :<|> "check-backup" :> Capture "backup-name" FilePath :> Get '[PlainText] Backup.BackupStatus
```

Comment:
I am not sure about the proliferation of "unversioned" endpoints. We have already `health-check`, `info`, and `config`. I think one can come up with reasons why the former are unversioned. I am sure about the backup endpoint. I think, what do you think about putting it under `chainweb/0.0/<network>/`?

Reply:
How long does a backup usually take? Could it make sense to just provide a single synchronous call?

Also, what happens if the user issues overlapping backup requests?

Reply:
Way too long for a synchronous call, because of sqlite.

Overlapping backup requests are fine, they interact with different subtrees of the filesystem and can interleave safely because the VACUUM calls go through the PactQueue.